File: tests/fixtures/rust/src/net/conn.rs
//! A second connection layer, exporting a function with the same name as
//! `db::conn::open`. That ambiguity is the whole point: tier A cannot decide
//! which `open` a caller in a third file means, and gives up rather than
//! guessing. A compiler can.

/// Open a socket.
pub fn open(url: &str) -> bool {
    url.starts_with("tcp://")
}

File: tests/fixtures/typescript/net/conn.ts
/**
 * A second connection layer, exporting a function with the same name as
 * db/conn's. That ambiguity is the whole point: tier A cannot decide which
 * `open` a caller in a third file means, and gives up rather than guessing.
 */
export function open(url: string): boolean {
  return url.startsWith("tcp://");
}
